projects/backend/fronts.ts

```
@@ -3,52 +3,73 @@ import fromEntries from 'object.fromentries'
-    const articleFragments = fromEntries(
-        articleFragmentList.map(
-            (fragment): [string, NestedArticleFragment] => [
-                fragment.id,
-                fragment,
-            ],
-        ),
-    )
+    if (hasFailed(deserialized))
+        return withFailureMessage(deserialized, 'Response was not valid json')
+
+    const collection = deserialized as CollectionFromResponse
 
+    const articleFragmentList = collection.live.map(
+        (fragment): [number, NestedArticleFragment] => [
+            parseInt(fragment.id.replace('internal-code/page/', '')),
+            fragment,
+        ],
+    )
+    const articleFragments = fromEntries(articleFragmentList)
+    const ids: number[] = articleFragmentList.map(([id]) => id)
     const preview = live ? undefined : true
-    const capiArticles = await getArticles(Object.keys(articleFragments))
-    const articles: [string, Article][] = Object.entries(capiArticles).map(
-        ([key, article]) => {
-            const fragment =
-                articleFragments[`internal-code/page/${article.id}`] ||
-                articleFragments[key]
+    const [capiPrintArticles, capiSearchArticles] = await Promise.all([
+        attempt(getArticles(ids, 'printsent')),
+        attempt(getArticles(ids, 'search')),
+    ])
+    if (hasFailed(capiPrintArticles))
+        return withFailureMessage(
+            capiPrintArticles,
+            'Could not connect to capi print sent',
+        )
+    if (hasFailed(capiSearchArticles))
+        return withFailureMessage(
+            capiSearchArticles,
+            'Could not connect to CAPI',
+        )
+
+    const articles: [string, Article][] = Object.entries(articleFragments)
+        .filter(([key]) => {
+            const inResponse =
+                key in capiPrintArticles || key in capiSearchArticles
+            if (!inResponse) {
+                console.warn(`Removing ${key} as not in CAPI response.`)
```

Long term do we want to do this, it seems like a silent failure case ... not sure whether it's worth worrying about but just a thought?